cmd/csi-attacher/main.go

```diff
@@ -279,6 +280,18 @@ func main() {
 		*reconcileSync,
 	)
 
+	// handle SIGTERM and SIGINT by cancelling the context.
+	shutdownCtx, cancel := context.WithCancel(ctx)
+	shutdownHandler := server.SetupSignalHandler()
+	ctx, terminate := context.WithCancel(shutdownCtx)
+	defer terminate()
+
+	go func() {
+		defer cancel()
+		<-shutdownHandler
```

Comment:
Does this work on Windows nodes too? SIGTERM behaves a bit different in Windows than Linux?

Reply:
Good cache! I think on windows it does os.interupt ([here](https://github.com/kubernetes/kubernetes/blob/34e80be1330556848521314ece92e9277efb26d4/staging/src/k8s.io/apiserver/pkg/server/signal_windows.go#L23) is how kubernetes handles it AFAIK) and just kills the process, I am not sure if we can cache this signal... I will test this with the updated code (I should update the PR by Wednesday based on previous comments). If I cannot think of a way to make this work for Windows we can discuss how we want to handle it.

Reply:
sorry for the delay, I have some issues with testing the code and cluster creation... I will update once I handle these issues 😅

Reply:
@sunnylovestiramisu Kubernetes does not ship any Windows containers with the CSI sidecars. We build just the Linux containers ([here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-sig-storage.yaml))

Vendors, e.g. Microsoft, ships the driver registrar and liveness probe (see e.g. [here](https://github.com/kubernetes-csi/csi-driver-smb/blob/475af8ac63f67cc0e40daba5a0d07717e92e1148/deploy/csi-smb-node-windows-hostprocess.yaml#L45C18-L45C88)), but even them don't seem to ship the Windows version of the attacher, provisioner, resizer or snapshotter.

Do we really want to support these sidecars on Windows? Nobody has probably run them there.

Reply:
I asked on [slack](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1744723105562959), even Microsoft runs the sidecars on Linux